ethcore/src/client/traits.rs

```diff
@@ -378,6 +378,9 @@ pub trait BlockChainClient : Sync + Send + AccountData + BlockChain + CallContra
 	/// Schedule state-altering transaction to be executed on the next pending block.
 	fn transact_contract(&self, address: Address, data: Bytes) -> Result<(), transaction::Error>;
 
+	/// Call contract as block author
+	fn call_contract_as_author(&self, id: BlockId, address: Address, data: Bytes) -> Result<Bytes, String>;
```

Comment:
Not sure I like this addition. Seems like a convenience method, but maybe I'm missing something. I'm hesitant to add new trait methods if we can avoid it.

Reply:
The issue of not using `call_contract_from_sender` is because we can't get blocks author outside client in `transact` method of `ValidatorContract`. By avoiding usage of this method we would need to introduce getter for author or the whole miner instance (which is available but only for tests):  

https://github.com/paritytech/parity-ethereum/blob/af932e6806b080f50d1f3d83c0e96348a6e91e84/ethcore/src/client/client.rs#L1001

Honestly I'm not sure what would be the best solution here. @dvdplm any thoughts about that?

Reply:
Not sure either tbh. @tomusdrw?